pkg/webhook/mutation/pod/init_container.go

```diff
@@ -44,7 +44,7 @@ func initContainerResources(dk dynakube.DynaKube) corev1.ResourceRequirements {
 		return *customInitResources
 	}
 
-	if !dk.OneAgent().IsCSIAvailable() {
+	if !dk.OneAgent().IsCSIAvailable() && !dk.MetadataEnrichmentEnabled() {
```

Comment:
this is not fully correct

scenarios you want to cover:
- no-csi + metadata + oa ==> no limit (unless custom is set)
  - why?: we have to download the archive and unzip it, the resource limits will not work 
- no-csi + metadata only ==> default limit
- no-csi + oa only ==> no limit
- csi + anything ==> default limit (unless custom is set `dk.OneAgent().GetInitResources()`)

Reply:
Thanks for the feedback!

I think I covered all cases now: https://github.com/Dynatrace/dynatrace-operator/pull/4288/commits/72b3d43ee7600d36c9889ba01fbc5a96787764ef